src/pages/index.js

```diff
@@ -57,42 +106,146 @@ const closeButtons = document.querySelectorAll(".modal__close");
 const userInfo = new UserInfo({
   nameSelector: ".profile__title",
   jobSelector: ".profile__description",
+  avatarSelector: ".profile__image",
+});
+
+const profileImagePopup = new PopupWithForm({
+  popupSelector: "#profile-image-modal",
+  handleFormSubmit: handleProfileImageSubmit,
+  formSelector: "#modal-image-form",
 });
 
+const deleteCardPopup = new DeleteConfirmation("#delete-conformation-modal");
+deleteCardPopup.setEventListeners();
+
+// Pass this to Card constructor
+function handleDeleteClick(card) {
+  deleteCardPopup.open();
+  // Call the popup's open method
+  // Call the setSubmitAction method and pass it an anonymous function
+  deleteCardPopup.setSubmitAction(() => {
+    console.log("card deletion handler called");
+    // call the api function (delete handler)
+    //    - pass it the id
+    //    - after successful response delete the card
+    let id = card.getId();
+    api.deleteCard(id).then(() => {
+      card.removeCard();
+      deleteCardPopup.close();
+      console.log("It ran");
+    });
```

Comment:
You need to `catch` possible errors at the very **end** of any server request.

**Please, correct it everywhere in the code**